### PR TITLE
fix: Really postselect particles in `addGeant4`

### PR DESCRIPTION
**Pull request by benjaminhuth**
The postselection used so far the configuration of the preselection.

Also, so far the post selection did not affect the particle writing. People who apply post-selection probably also want to export only the selected particles.

This PR also applies the post-selection to "particles_final", since it was only applied to "particles_initial" so far.

### CODE REVIEW

File: Examples/Python/python/acts/examples/simulation.py [hunk 1]
```diff
@@ -625,16 +627,27 @@ def addGeant4(
...
         addParticleSelection(
             s,
-            preSelectParticles,
+            postSelectParticles,
             inputParticles=g4conf.outputParticlesInitial,
-            outputParticles=postSelectedParticles,
+            outputParticles=particlesInitial,
+        )
+
+        particlesFinal = "particles_final_selected"
```
Comment (andiwand) [inline Examples/Python/python/acts/examples/simulation.py:638]: what is the use case for final particles selected?

File: Examples/Python/python/acts/examples/simulation.py [hunk 2]
```diff
@@ -625,16 +627,27 @@ def addGeant4(
...
+            inputParticles=g4conf.outputParticlesFinal,
+            outputParticles=particlesFinal,
         )
-        s.addWhiteboardAlias("particles", postSelectedParticles)
     else:
-        s.addWhiteboardAlias("particles", alg.config.outputParticlesInitial)
+        particlesInitial = alg.config.outputParticlesInitial
+        particlesFinal = alg.config.outputParticlesFinal
+
+    # Only add alieas for 'particles_initial' as this is the one we use most
```
Comment (andiwand) [inline Examples/Python/python/acts/examples/simulation.py:649]:
```suggestion
    # Only add alias for 'particles_initial' as this is the one we use most
```

File: Examples/Python/python/acts/examples/simulation.py [hunk 3]
```diff
@@ -625,16 +627,27 @@ def addGeant4(
...
+    s.addWhiteboardAlias("particles", particlesInitial)
```
Comment (andiwand) [inline Examples/Python/python/acts/examples/simulation.py:650]: I think we should reflect all these changes in `addFatras` too

File: Examples/Python/python/acts/examples/simulation.py (hunk 1 above)
Comment (benjaminhuth) [reply Examples/Python/python/acts/examples/simulation.py:638]: mainly that we write them out. In my case it makes quite a difference (over 100MB per event).

File: Examples/Python/python/acts/examples/simulation.py (hunk 3 above)
Comment (benjaminhuth) [reply Examples/Python/python/acts/examples/simulation.py:650]: yeah good point, I'll update it later today

File: Examples/Python/python/acts/examples/simulation.py (hunk 1 above)
Comment (andiwand) [reply Examples/Python/python/acts/examples/simulation.py:638]: that't a good point I didn't think of that :+1:

File: Examples/Python/python/acts/examples/simulation.py (hunk 3 above)
Comment (asalzburger) [reply Examples/Python/python/acts/examples/simulation.py:650]: Was this done?

**Review by andiwand — APPROVED** [review]
lets gooo

**Pull request merged**